File: MiniDOMBase/DMD/Sistema/Modules/ModuleEventArgs.vb
Imports System.Net.Mail 'importo il Namespace
Imports System.Net.Sockets
Imports minidom
Imports minidom.Databases
Imports minidom.Anagrafica

Partial Public Class Sistema


    Public Class ModuleEventArgs
        Inherits System.EventArgs

        Private m_Module As CModule

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
        End Sub

        Public Sub New(ByVal m As CModule)
            Me.New()
            Me.m_Module = m
        End Sub

        Public ReadOnly Property [Module] As CModule
            Get
                Return Me.m_Module
            End Get
        End Property

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class


End Class

File: MiniDOMBase/DMD/Sistema/Modules/Events/ModuleEventArgs.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;

namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Evento relativo ad un modulo
        /// </summary>
        [Serializable]
        public class ModuleEventArgs
            : DMDEventArgs
        {
            
            [NonSerialized] private CModule m_Module;

            /// <summary>
            /// Costruttore
            /// </summary>
            public ModuleEventArgs()
            {
                //DMDObject.IncreaseCounter(this);
            }

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="m"></param>
            public ModuleEventArgs(CModule m) : this()
            {
                m_Module = m;
            }

            /// <summary>
            /// Restituisce un riferimento al modulo
            /// </summary>
            public CModule Module
            {
                get
                {
                    return m_Module;
                }
            }

            //~ModuleEventArgs()
            //{
            //    //DMDObject.DecreaseCounter(this);
            //}
        }
    }
}